src/nvim/option.c

```diff
@@ -900,11 +900,18 @@ set_option_default(
       if (options[opt_idx].indir == PV_SCROLL) {
         win_comp_scroll(curwin);
       } else {
-        *(long *)varp = (long)(intptr_t)options[opt_idx].def_val[dvi];
+        long def_val = (long)options[opt_idx].def_val[dvi];
+        if ((long *)varp == &curwin->w_p_so
+                || (long *)varp == &curwin->w_p_siso)
```

Comment:
`if` should always use braces (the linter will tell you about the relevant cases)